packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryBuffer.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+// Default duration threshold for reporting performance entries (0 means "report
+// all")
+constexpr double DEFAULT_DURATION_THRESHOLD = 0.0;
+
+/**
+ * Abstract performance entry buffer with reporting flags.
+ * Subtypes differ on how entries are stored.
+ */
+class PerformanceEntryBuffer {
+ public:
+  double durationThreshold{DEFAULT_DURATION_THRESHOLD};
+  size_t droppedEntriesCount{0};
+
+  explicit PerformanceEntryBuffer() = default;
+  virtual ~PerformanceEntryBuffer() = default;
+
+  virtual void add(const PerformanceEntry& entry) = 0;
+  virtual void getEntries(
+      std::optional<std::string_view> name,
```

Comment:
Nit: if the name is optional then it's probably better to have 2 different methods (`getEntries(target)` and `getEntries(name, target)`).